src/main/java/org/mule/extension/db/internal/domain/connection/DefaultDbConnection.java

```diff
@@ -7,27 +7,58 @@
 
 package org.mule.extension.db.internal.domain.connection;
 
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
 import static java.util.Collections.emptyList;
+import static org.mule.extension.db.api.param.JdbcType.BLOB;
+import static org.mule.extension.db.api.param.JdbcType.CLOB;
 import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.core.api.util.IOUtils.toByteArray;
 import org.mule.extension.db.api.exception.connection.ConnectionClosingException;
+import org.mule.extension.db.internal.domain.connection.type.resolver.CollectionTypeResolver;
+import org.mule.extension.db.internal.domain.connection.type.resolver.StructTypeResolver;
+import org.mule.extension.db.internal.domain.connection.type.resolver.TypeResolver;
 import org.mule.extension.db.internal.domain.type.DbType;
+import org.mule.extension.db.internal.domain.type.ResolvedDbType;
 import org.mule.extension.db.internal.result.resultset.ResultSetHandler;
 import org.mule.extension.db.internal.result.statement.GenericStatementResultIteratorFactory;
 import org.mule.extension.db.internal.result.statement.StatementResultIteratorFactory;
 import org.mule.runtime.api.tx.TransactionException;
+import org.mule.runtime.core.api.util.IOUtils;
 
+import java.io.InputStream;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
 import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.Struct;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class DefaultDbConnection implements DbConnection {
 
   private final Connection jdbcConnection;
   private final List<DbType> customDataTypes;
   private AtomicInteger streamsCount = new AtomicInteger(0);
   private boolean isTransactionActive = false;
 
+  public static final int DATA_TYPE_INDEX = 5;
+  public static final int ATTR_TYPE_NAME_INDEX = 6;
+  private static final List<String> LOB_TYPES = asList(BLOB.getDbType().getName(), CLOB.getDbType().getName());
+
+  protected static final int UNKNOWN_DATA_TYPE = -1;
+
+  protected Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
loggers are expensive to create. Make this `static final`

Reply:
'this cannot be referenced from a static context' I change it to final, is that enough?

Reply:
no is not enough. Don't use getClass(), just use `DefaultDbConnection.class`